sklearn/manifold/tests/test_t_sne.py

```diff
@@ -277,6 +278,7 @@ def test_fit_csr_matrix(method):
                     1.0, rtol=1.1e-1)
 
 
+@ignore_warnings(category=FutureWarning)
```

Comment:
Why not update the tests to use `square_distance=True`?

That way, when the API changes, these tests will already be correct--no need to remove a bunch of `ignore_warnings` pragmas :).

You can also add additional happy-path tests here to exercise the code paths where `square_distance` is `False` or `'legacy'`.

Reply:
Ignore the last paragraph; I see you already added those tests.

Reply:
My thinking was that I wanted to make minimal changes to the bodies of the tests themselves in this PR. 

Setting square_distance=True would change the behavior of those tests to the future functionality. I wanted to delay that change until the default is truly updated in 0.26.

Reply:
I agree that we need tests for true, false and legacy. However for the tests that only check that proper errors are raised for wrong parameters we can safely switch to the new behavior.

Reply:
Also it would be good to make sure that all tests still pass with squared_distances = True

Reply:
Thank you for the input! For now, I've made this change by parametrizing `square_distance` to check `True` and `False` across relevant tests. But, as was mentioned elsewhere, I have mixed feelings about the `square_distance=False` behavior in general.